### ISSUE TITLE
[DOC] Improve docs of `evaluate` re splits

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#5468]
The docstring of `evaluate` should be improved so at the start it makes a note about time splits, known/unknown vars, and possible leakage. See https://github.com/sktime/sktime/issues/5438

FYI @admo1

**Comment by adamkells** [comment]
@fkiraly Will take this one.

**Comment by adamkells** [comment]
Thinking about this raised a very basic question for me. When we forecast with exogenous variables, do we use the value of X aligned with the time we are making the prediction or the time we are predicting?

If we are forecasting for 2001, 2002, and 2003, should the exogenous variables (X) provided during the fitting process (fit method) correspond to the values observed in 2001, 2002, and 2003, or should they align with the time points we are actually predicting?

For example, if we are predicting the value for 2001, should the exogenous variables at that prediction step be the ones observed in 2001, or should they be the values observed at the time of model fitting (which may be earlier)? 

I think the darts package disambiguates this with past_covariates and future_covariates. If I understand correctly, does sktime just use the equivalent of future_covariates and require some manual lagging to recreate past_covariates?

**Comment by fkiraly** [comment]
> If I understand correctly, does sktime just use the equivalent of future_covariates and require some manual lagging to recreate past_covariates?

No, the `sktime` design is different - non-nan values are assumed known. The model is expected to deal with known/unknown for the future, if `X` is passed in `predict`, non-nan values are assumed known.

This was a conscious decision to keep the method signatures simple.
